openshift-ci/scripts/oci-model-registry-deploy.sh

```diff
@@ -1,40 +1,175 @@
+            # Get the pod info
+            local pod_info=$(oc get pod "$pod_name" -n "$namespace" --no-headers)
+
+            # Extract pod status and ready status from the info
+            local pod_status=$(echo "$pod_info" | awk '{print $3}')
+            local pod_ready=$(echo "$pod_info" | awk '{print $2}')
+            local ready_containers=$(echo "$pod_ready" | cut -d'/' -f1)
+
+            # Check if the pod is Running and all containers are ready
+            if [[ $pod_status == "Running" ]] && [[ $ready_containers -eq $expected_ready_containers ]]; then
+                echo -e "${GREEN}✔ Success:${NC} Pod $pod_name is running and $ready_containers out of $expected_ready_containers containers are ready"
+                return 0  # Success
+            else
+                echo -e "${YELLOW}! Info:${NC} Pod $pod_name is not running or does not have $expected_ready_containers containers ready"
+            fi
+        done <<< "$pod_list"
+
+        sleep 5  # Wait for 5 seconds before checking again
+    done
+
+    echo -e "${RED}X Failure:${NC} Timeout reached. No pod matching '$pod_name_partial' became ready within $timeout seconds"
+    return 1  # Failure
+}
+
+check_route_status() {
+    local namespace="$1"
+    local route_name="$2"
+    local key="items"
+    local interval=5
+    local timeout=300
+    local start_time=$(date +%s)
+
+    while (( $(date +%s) - start_time < timeout )); do
+        # Get the route URL
+        local route=$(oc get route -n "$namespace" "$route_name" -o jsonpath='{.spec.host}')
+        local route_url="http://$route"
+
+        if [[ -z "$route_url" ]]; then
+             echo -e "${RED}X Fail:${NC} Route '$route_name' does not exist in namespace '$namespace'"
+            return 1
+        else 
+            echo -e "${GREEN}✔ Success:${NC} Route '$route_name' exists in namespace '$namespace'"
+        fi
+
+        # Test if the route is live
+        local response=$(curl -s -o /dev/null -w "%{http_code}" "$route_url/api/model_registry/v1alpha2/registered_models")
+
+        # Check if the response status code is 200 OK or 404 Not Found
+        if [[ "$response" == "200" ]]; then
+            echo -e "${GREEN}✔ Success:${NC} Route server is reachable. Status code: 200 OK"
+            return 0
+        elif [[ "$response" == "404" ]]; then
+            echo -e "${GREEN}✔ Success:${NC} Route server is reachable. Status code: 404 Not Found"
+            return 0
+        else
+            echo -e "${RED}X Fail:${NC} Route server is unreachable. Status code: $response"
+        fi
+
+        sleep "$interval"
+    done
+
+    echo -e "${RED}X Fail:${NC} Timeout reached. Route '$route_name' did not become live within $timeout seconds."
+    return 1
+}
+
+# Run the deployment tests.
+run_deployment_tests() {
+    check_deployment_availability default model-registry-db
+    check_deployment_availability default modelregistry-sample
+    check_pod_status default model-registry 1
```

Comment:
isn't this pod called `model-registry-db` ? 🤔

Reply:
Are you referring to line 160?  In line 87 the function takes the unique partial name of the pod and returns a list of matching pods. I was just trying to build in a little bit of redundancy and not be too specific.

Reply:
> Are you referring to line 160? 

yes.

> In line 87 the function takes the unique partial name of the pod(s) and returns a list of matching pods. I was just trying to build in a little bit of redundancy and not be too specific.

okay thank for clarification. I believe there are k8s functions that instead of grepping for the name prefix, you can query pod for `app=modelregistry-sample` or `component=model-registry` or the likes, I _believe_ it would make this test logic more robust than grep-ping, but up to you.

Thanks again for the clarification

Reply:
I have tried to run the script using these suggestions and various other options using selectors, however they do not work.  I will leave it as it is for now and do some more research on the best method.  The solution might be to create labels for the apps that can be used to filter.

Reply:
I have changed this to use selectors instead of grep